packages/flutter/test/material/dropdown_test.dart

```diff
@@ -1283,4 +1292,234 @@ void main() {
+  testWidgets('Dropdown button form field with isDense:true aligns selected menu item', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+    const String value = 'two';
+
+    await tester.pumpWidget(
+      buildFormFrame(
+        buttonKey: buttonKey,
+        value: value,
+        isDense: true,
+        onChanged: onChanged
+      ),
+    );
+    final RenderBox buttonBox = tester.renderObject<RenderBox>(
+      find.byKey(buttonKey),
+    );
+    assert(buttonBox.attached);
+
+    await tester.tap(find.text('two'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu animation
+
+    // The selected dropdown item is both in menu we just popped up, and in
+    // the IndexedStack contained by the dropdown button. Both of them should
+    // have the same vertical center as the button.
+    final List<RenderBox> itemBoxes = tester.renderObjectList<RenderBox>(
+      find.byKey(const ValueKey<String>('two')),
+    ).toList();
+    expect(itemBoxes.length, equals(2));
+
+    // When isDense is true, the button's height is reduced. The menu items'
+    // heights are not.
+    final List<double> itemBoxesHeight = itemBoxes.map<double>((RenderBox box) => box.size.height).toList();
+    final double menuItemHeight = itemBoxesHeight.reduce(math.max);
+    expect(menuItemHeight, greaterThanOrEqualTo(buttonBox.size.height));
+
+    for (RenderBox itemBox in itemBoxes) {
+      expect(itemBox.attached, isTrue);
+      final Offset buttonBoxCenter = buttonBox.size.center(buttonBox.localToGlobal(Offset.zero));
+      final Offset itemBoxCenter = itemBox.size.center(itemBox.localToGlobal(Offset.zero));
+      expect(buttonBoxCenter.dy, equals(itemBoxCenter.dy));
+    }
+  });
+
+  testWidgets('Dropdown button form field changes style when specified', (WidgetTester tester) async {
+    const String value = 'foo';
+    final UniqueKey itemKey = UniqueKey();
+
+    await tester.pumpWidget(
+      TestApp(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: DropdownButtonFormField<String>(
+            value: value,
+            items: <DropdownMenuItem<String>>[
+              DropdownMenuItem<String>(
+                key: itemKey,
+                value: 'foo',
+                child: const Text(value),
+              ),
+            ],
+            isDense: true,
+            onChanged: (_) { },
+            style: const TextStyle(
+              color: Colors.amber,
+              fontSize: 20.0,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final RichText richText = tester.widget<RichText>(
+      find.descendant(
+        of: find.byKey(itemKey),
+        matching: find.byType(RichText),
+      ),
+    );
+
+    expect(richText.text.style.color, Colors.amber);
+    expect(richText.text.style.fontSize, 20.0);
+  });
+
+  testWidgets('disabledHint displays on empty items or onChanged for form field', (WidgetTester tester) async {
```

Comment:
```suggestion
  testWidgets('Dropdown form field - disabledHint displays when the items list is empty, when items is null, or when onChanged is null', (WidgetTester tester) async {
```

Reply:
This doesn't seem to test for when [onChanged] is null, which makes this test misleading. I would add it as a separate unit test

Reply:
onChanged is handled by formfieldstate itself. Si i guess no test is needed for it.